azext_iot/iothub/providers/state.py

```diff
@@ -0,0 +1,640 @@
+            if len(hub_certs) < len(hub_state["arm"]["resources"]) - 1:
+                logger.warning("Private endpoints for IoT Hub will be ignored for state import.")
+
+            for res in hub_certs:
+                res["name"] = self.hub_name + "/" + res["name"].split("/")[1]
+                depends_on = res["dependsOn"][0].split("'")
+                depends_on[3] = self.hub_name
+                res["dependsOn"][0] = "'".join(depends_on)
+            hub_resources.extend(hub_certs)
+            hub_state["arm"]["resources"] = hub_resources
+
+            filename = f"arm_deployment{self.hub_name}.json"
+            with open(filename, "w", encoding='utf-8') as f:
+                json.dump(hub_state["arm"], f)
+
+            arm_result = cli.invoke(
+                f"deployment group create --template-file {filename} -g {self.rg}"
+            )
+            os.remove(filename)
+
+            if not arm_result.success():
+                raise BadRequestError(f"Arm deployment for IoT Hub {self.hub_name} failed.")
+
+            if not self.target:
+                self.target = self.discovery.get_target(
+                    hub_resource["name"],
+                    resource_group_name=arm_result.as_json()["resourceGroup"]
+                )
+                print(f"Created IoT Hub {self.hub_name}.")
+            else:
+                print(f"Updated IoT Hub {self.hub_name}.")
+
+        # Data plane
+        # upload configurations
+        if HubAspects.Configurations.value in hub_aspects and hub_state.get("configurations"):
+            hub_aspects.remove(HubAspects.Configurations.value)
+            configs = hub_state["configurations"]["admConfigurations"]
+            for config_id, config_obj in tqdm(configs.items(), desc="Uploading ADM Configurations"):
+                try:
+                    _iot_hub_configuration_create(
+                        target=self.target,
+                        config_id=config_id,
+                        content=json.dumps(config_obj["content"]),
+                        target_condition=config_obj["targetCondition"],
+                        priority=config_obj["priority"],
+                        labels=json.dumps(config_obj["labels"]),
+                        metrics=json.dumps(config_obj["metrics"])
+                    )
+                except AzCLIError as e:
+                    logger.error(f" Failed to upload ADM configuration {config_id}. Error Message: {e}")
+
+            edge_deployments = hub_state["configurations"]["edgeDeployments"]
+            for config_id, config_obj in tqdm(edge_deployments.items(), desc="Uploading Edge Deployments"):
+                if "properties.desired" in config_obj["content"]["modulesContent"]["$edgeAgent"]:
+                    config_type = ConfigType.edge
+                else:
+                    config_type = ConfigType.layered
```

Comment:
At a minimum we should apply layered deployments after all the full edge deployments. There could be more logic needed for a complete solution where one layered deployments depends on another, but we can revisit when the need arises later.